reactor-core/src/main/java/reactor/core/publisher/FluxSwitchOnFirst.java

```diff
@@ -19,610 +19,1029 @@
+				boolean cancelSourceOnComplete) {
+			this.outer = outer instanceof Fuseable.ConditionalSubscriber ?
+					new SwitchOnFirstConditionalControlSubscriber<>(this,
+							(Fuseable.ConditionalSubscriber<R>) outer,
+							cancelSourceOnComplete) :
+					new SwitchOnFirstControlSubscriber<>(this,
+							outer,
+							cancelSourceOnComplete);
+			this.transformer = transformer;
+		}
+
+		@Override
+		@Nullable
+		public Object scanUnsafe(Attr key) {
+			final boolean isCancelled = this.inner == Operators.EMPTY_SUBSCRIBER;
+
+			if (key == Attr.CANCELLED) {
+				return isCancelled && !this.done;
+			}
+			if (key == Attr.TERMINATED) {
+				return this.done || isCancelled;
+			}
+			if (key == Attr.RUN_STYLE) {
+				return Attr.RunStyle.SYNC;
+			}
+
+			return InnerOperator.super.scanUnsafe(key);
+		}
+
+		@Override
+		public CoreSubscriber<? super R> actual() {
+			return this.outer;
+		}
+
+		@Override
+		public void onSubscribe(Subscription s) {
+			if (Operators.validate(this.s, s)) {
+				this.s = s;
+				this.outer.sendSubscription();
+				if (!hasInboundCancelled(this.state)) {
+					s.request(1);
+				}
+			}
+		}
+
+		@Override
+		public void onNext(T t) {
+			if (this.done) {
+				Operators.onNextDropped(t, currentContext());
+				return;
+			}
+
+			final T f = this.firstValue;
+			if (f == null) {
+				this.firstValue = t;
+
+				final long previousState = markFirstValueReceived(this);
+				if (hasInboundCancelled(previousState)) {
+					Operators.onDiscard(t, this.outer.currentContext());
+					return;
+				}
+
+				final Publisher<? extends R> result;
+				final CoreSubscriber<? super R> o = this.outer;
+
+				try {
+					result = Objects.requireNonNull(
+						this.transformer.apply(Signal.next(t, o.currentContext()), this),
+						"The transformer returned a null value"
+					);
+				}
+				catch (Throwable e) {
+					this.done = true;
+					o.onError(Operators.onOperatorError(this.s, e, t, o.currentContext()));
+					return;
+				}
+
+				result.subscribe(o);
+				return;
+			}
+
+			this.inner.onNext(t);
+		}
+
+		@Override
+		public void onError(Throwable t) {
+			if (this.done) {
+				Operators.onErrorDropped(t, this.outer.currentContext());
+				return;
+			}
+
+			this.done = true;
+			this.throwable = t;
+
+			final long previousState = markInboundTerminated(this);
+			if (hasInboundCancelled(previousState) || hasInboundTerminated(previousState)) {
+				Operators.onErrorDropped(t, this.outer.currentContext());
+				return;
+			}
+
+			if (hasFirstValueSent(previousState)) {
+				this.inner.onError(t);
+				return;
+			}
+
+			if (!hasFirstValueReceived(previousState)) {
+				final Publisher<? extends R> result;
+				final CoreSubscriber<? super R> o = this.outer;
+				try {
+					result = Objects.requireNonNull(this.transformer.apply(Signal.error(t,
+							o.currentContext()), this),
+							"The transformer returned a null value");
+				}
+				catch (Throwable e) {
+					o.onError(Exceptions.addSuppressed(e, t));
```

Comment:
Since the only guidance for suppressed exceptions that I know of is basically "the suppressing exception is considered more prominent/relevant than the suppressed exception", I wonder if it would be better to reverse and make the `onError` more prominent?

```java
Exceptions.addSuppressed(t, e)
```
Not a very strong opinion on my part, but it just made me wonder what you thought.